### PR TITLE
add summary and testimonial card documentation

### PR DESCRIPTION
**Pull request by wibjorn**
Task: task-695992
Link: preview-453

https://design.docs.microsoft.com/pulls/453/patterns/card.html

Adds a section in patterns/card for two cards. Summary card and testimonial card. Also renames resource card to icon card.
## Testing

1. Read through page above.
2. Use example markup to test grid set up.

Markup for testing grids of testimonial card

```html
<div class="display-grid gap-md" style="grid-template-columns: 1fr 1fr">
	<article class="card">
	<div class="card-content font-size-lg display-flex flex-direction-column">
		<div class="media">
			<div class="media-left">
				<div class="image image-64x64">
					<img class="border-radius-rounded" src="~/src/scaffold/media/twitter-example.jpeg" />
				</div>
			</div>
			<div class="media-content">
				<p class="font-size-xl font-weight-semibold margin-bottom-xxs">Full Name</p>
				<p href="#">Role</p>
			</div>
		</div>
		<p class="margin-block-xs color-text-subtle">
			"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor." "Lorem
				ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor." "Lorem ipsum
				dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor." "Lorem ipsum dolor sit
				amet, consectetur adipiscing elit, sed do eiusmod tempor."
		</p>
		<a href="#" class="display-block text-decoration-none margin-top-auto font-size-md"
			>Read this person's story</a
		>
	</div>
</article>
	<article class="card">
	<div class="card-content font-size-lg display-flex flex-direction-column">
		<div class="media">
			<div class="media-left">
				<div class="image image-64x64">
					<img class="border-radius-rounded" src="~/src/scaffold/media/twitter-example.jpeg" />
				</div>
			</div>
			<div class="media-content">
				<p class="font-size-xl font-weight-semibold margin-bottom-xxs">Full Name</p>
				<p href="#">Role</p>
			</div>
		</div>
		<p class="margin-block-xs color-text-subtle">
			"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor."
		</p>
		<a href="#" class="display-block text-decoration-none margin-top-auto font-size-md"
			>Read this person's story</a
		>
	</div>
</article>
</div>
```

Markup for testing grids of Icon cards

```html
<div class="display-grid gap-md" style="grid-template-columns: 1fr 1fr">
	<article class="card position-relative">
		<div class="card-content display-flex">
			<div class="media flex-grow-1 align-items-stretch">
				<div class="media-left">
					<div class="image image-64x64">
						<img src="~/src/scaffold/media/video-media.png" />
					</div>
				</div>
				<div class="media-content display-flex flex-direction-column align-items-space-between">
					<div>
						<p class="text-transform-uppercase margin-bottom-xxs">Videos</p>
						<p class="font-size-lg font-weight-semibold margin-bottom-xxs">
							Media content is a place to put your text content.
						</p>
						<p class="color-text-subtle margin-bottom-xs">
							A description can go here. A description can go here. A description can go here. A
							description can go here.
						</p>
					</div>
					<a class="text-decoration-none" href="#">Watch videos</a>
				</div>
			</div>
		</div>
	</article>
	<article class="card position-relative">
		<div class="card-content display-flex">
			<div class="media flex-grow-1 align-items-stretch">
				<div class="media-left">
					<div class="image image-64x64">
						<img src="~/src/scaffold/media/video-media.png" />
					</div>
				</div>
				<div class="media-content display-flex flex-direction-column align-items-space-between">
					<div>
						<p class="text-transform-uppercase margin-bottom-xxs">Videos</p>
						<p class="font-size-lg font-weight-semibold margin-bottom-xxs">
							Media content is a place to put your text content.
						</p>
						<p class="color-text-subtle margin-bottom-xs">Videos to help you get started.</p>
					</div>
					<a class="text-decoration-none justify-self-stretch margin-top-auto" href="#"
						>Watch videos</a
					>
				</div>
			</div>
		</div>
	</article>
</div>
```

### CODE REVIEW

File: site/src/patterns/card.md [hunk 1]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
-				<p class="font-size-lg font-weight-semibold margin-bottom-xxs">
-					Media content is a place to put your text content.
-				</p>
-				<p class="color-text-subtle margin-bottom-xs">Videos to help you get started.</p>
-				<a class="text-decoration-none stretched-link" href="#">Watch videos</a>
+				<p class="font-size-xl font-weight-semibold margin-bottom-xxs">First Last Name</p>
+				<p href="#">Role</p>
 			</div>
 		</div>
+		<p class="margin-block-xs">
```
Comment (ollips) [inline site/src/patterns/card.md:118]: Please add `color-text-subtle` to match the design.

File: site/src/patterns/card.md [hunk 2]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
+</article>
+```
+
+## Testimonial card
+
+The testimonial card provides a place for a user's opinion to be showcased. Notable modifications from Icon Card include using `border-radius-rounded` on the image, as well as slightly different font and text sizes. There are also several flexbox based atomics applied to ensure links justify at the ends of cards when placed in a grid.
+
+```html
+<article class="card">
+	<div class="card-content font-size-lg display-flex flex-direction-column">
```
Comment (ollips) [inline site/src/patterns/card.md:106]: Looks good in grid 🙌🏻

image

File: site/src/patterns/card.md [hunk 3]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
+			truncated the more we repeat this sentence. The longer description of the card, which will
+			eventually become truncated the more we repeat this sentence.
+		</p>
+	</div>
+</article>
+```
+
+## Testimonial card
+
+The testimonial card provides a place for a user's opinion to be showcased. Notable modifications from Icon Card include using `border-radius-rounded` on the image, as well as slightly different font and text sizes. There are also several flexbox based atomics applied to ensure links justify at the ends of cards when placed in a grid.
```
Comment (homing1) [inline site/src/patterns/card.md:102]: supernit: "for an user's opinion"

File: site/src/patterns/card.md [hunk 4]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
-				<p class="color-text-subtle margin-bottom-xs">Videos to help you get started.</p>
-				<a class="text-decoration-none stretched-link" href="#">Watch videos</a>
+				<p class="font-size-xl font-weight-semibold margin-bottom-xxs">First Last Name</p>
+				<p href="#">Role</p>
 			</div>
 		</div>
+		<p class="margin-block-xs">
+			"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor."
+		</p>
+		<a href="#" class="display-block text-decoration-none margin-top-auto"
```
Comment (ollips) [inline site/src/patterns/card.md:121]: `font-size-md`, per design should be smaller.

File: site/src/patterns/card.md (hunk 2 above)
Comment (homing1) [reply site/src/patterns/card.md:106]: What about in mobile? 
image

Comment (ollips) [reply site/src/patterns/card.md:106]: In mobile cards would stack one on top of the other ;) 

image

File: site/src/patterns/card.md (hunk 3 above)
Comment (BenDMyers) [reply site/src/patterns/card.md:102]: I think it is supposed to be _a user_, given _user_ starts with the _yoo_ sound and not an _oo_ sound.

File: site/src/patterns/card.md [hunk 5]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
+
+The testimonial card provides a place for a user's opinion to be showcased. Notable modifications from Icon Card include using `border-radius-rounded` on the image, as well as slightly different font and text sizes. There are also several flexbox based atomics applied to ensure links justify at the ends of cards when placed in a grid.
+
+```html
+<article class="card">
+	<div class="card-content font-size-lg display-flex flex-direction-column">
+		<div class="media position-relative">
+			<div class="media-left">
+				<div class="image image-64x64">
+					<img class="border-radius-rounded" src="~/src/scaffold/media/twitter-example.jpeg" />
```
Comment (BenDMyers) [inline site/src/patterns/card.md:110]: Is it worth providing an `alt` here, even if it's `alt=""`? That way, this code is a little safer to copy and paste?

Also, should we provide guidance about icon/avatar alt here, or would that be too prescriptive?

File: site/src/patterns/card.md [hunk 6]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
+				</div>
+			</div>
 			<div class="media-content">
-				<p class="text-transform-uppercase margin-bottom-xxs">Videos</p>
-				<p class="font-size-lg font-weight-semibold margin-bottom-xxs">
-					Media content is a place to put your text content.
-				</p>
-				<p class="color-text-subtle margin-bottom-xs">Videos to help you get started.</p>
-				<a class="text-decoration-none stretched-link" href="#">Watch videos</a>
+				<p class="font-size-xl font-weight-semibold margin-bottom-xxs">First Last Name</p>
```
[the remark this answers is not in the document]
Comment (BenDMyers) [reply site/src/patterns/card.md:114]: I might throw my hat in the ring for `Firstname Lastname`

File: site/src/patterns/card.md (hunk 2 above)
Comment (homing1) [reply site/src/patterns/card.md:106]: Nevermind, it has to do with the example markup. Ignore my comment please

File: css/src/atomics/spacing-auto.scss [hunk 7]
```diff
@@ -1,7 +1,8 @@
 $auto-spacing-properties: (
 	('margin-inline', 'margin-inline'),
 	('margin-left', 'margin-inline-start'),
-	('margin-right', 'margin-inline-end')
+	('margin-right', 'margin-inline-end'),
+	('margin-top', 'margin-block-start')
```
[the remark this answers is not in the document]
Comment (wibjorn) [reply css/src/atomics/spacing-auto.scss:5]: Updated, spacing doc. I don't really think that margin-bottom-auto or margin-block auto warrant adding at this time

File: site/src/patterns/card.md (hunk 3 above)
Comment (homing1) [reply site/src/patterns/card.md:102]: You're right, resolving

File: site/src/patterns/card.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (homing1) [reply site/src/patterns/card.md:114]: or just... `name`?

[the remark this answers is not in the document]
Comment (ollips) [reply site/src/patterns/card.md:114]: Anything but "First Last Name" would be fine 😁

File: site/src/patterns/card.md [hunk 8]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
 
-This card pattern makes use of the outer `.card` class and includes the use of the [`.media`](../components/media.md) and [`.image`](../components/image.md) components inside it. Also, because there is only one clickable element inside, it uses the [`.stretched-link` component](../components/stretched-link.md)
+This card pattern makes use of the outer `.card` class and includes the use of the [`.media`](../components/media.md) and [`.image`](../components/image.md) components inside it. Also, because there is only one clickable element inside, it uses the [`.stretched-link` component](../components/stretched-link.md). There are also several flexbox based atomics applied inside `.media` to ensure links justify at the ends of cards when placed in a grid.
 
 ```html
 <article class="card position-relative">
-	<div class="card-content">
-		<div class="media">
+	<div class="card-content display-flex">
+		<div class="media flex-grow-1 align-items-stretch">
```
Comment (wibjorn) [inline site/src/patterns/card.md:58]: can remove flex-grow-1 too

File: site/src/patterns/card.md (hunk 3 above)
Comment (wibjorn) [reply site/src/patterns/card.md:102]: Yes, vowel/consonant _sounds_ are what is important in article choice

File: site/src/patterns/card.md [hunk 9]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
+```
+
+## Testimonial card
+
+The testimonial card provides a place for a user's opinion to be showcased. Notable modifications from Icon Card include using `border-radius-rounded` on the image, as well as slightly different font and text sizes. There are also several flexbox based atomics applied to ensure links justify at the ends of cards when placed in a grid.
+
+```html
+<article class="card">
+	<div class="card-content font-size-lg display-flex flex-direction-column">
+		<div class="media position-relative">
```
[the remark this answers is not in the document]
Comment (wibjorn) [reply site/src/patterns/card.md:107]: nope, was a holdover from when there was a clickable twitter handle, oh design churn ...

File: site/src/patterns/card.md (hunk 2 above)
Comment (wibjorn) [reply site/src/patterns/card.md:106]: Yes, ideally the card doesn't change based on screensize, but the grid certainly would.

File: site/src/patterns/card.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (wibjorn) [reply site/src/patterns/card.md]: haha let's just do full name then :D

File: site/src/patterns/card.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (wibjorn) [reply site/src/patterns/card.md]: removed

File: site/src/patterns/card.md [hunk 10]
```diff
@@ -48,28 +48,79 @@ This card type contains a super title labelled by the card's type, a title, an i
...
 			</div>
+			<div class="media-content display-flex flex-direction-column align-items-space-between">
+				<div>
+					<p class="text-transform-uppercase margin-bottom-xxs">Videos</p>
+					<p class="font-size-lg font-weight-semibold margin-bottom-xxs">
+						Media content is a place to put your text content.
+					</p>
+					<p class="color-text-subtle margin-bottom-xs">Videos to help you get started.</p>
+				</div>
+				<a class="text-decoration-none justify-self-stretch margin-top-auto" href="#"
```
[the remark this answers is not in the document]
Comment (wibjorn) [reply site/src/patterns/card.md]: it's back

File: site/src/patterns/card.md (hunk 1 above)
Comment (wibjorn) [reply site/src/patterns/card.md]: updated

File: site/src/patterns/card.md (hunk 5 above)
Comment (wibjorn) [reply site/src/patterns/card.md:110]: Yes! I've added this to all images on this page, thanks for catching this

```
 alt="Write alt text here or if image is presentational only add aria-hidden attribute"

File: site/src/patterns/card.md [hunk 11]
```diff
@@ -48,28 +48,86 @@ This card type contains a super title labelled by the card's type, a title, an i
...
+		<div class="media align-items-stretch">
+			<div class="media-left">
+				<div class="image image-64x64">
+					<img
+						alt="Write alt text here or if image is presentational only add aria-hidden attribute"
+						src="~/src/scaffold/media/video-media.png"
+					/>
+				</div>
+			</div>
+			<div class="media-content display-flex flex-direction-column align-items-space-between">
```
Comment (homing1) [inline site/src/patterns/card.md:67]: It doesn't seem like we have `space-between` in our `align-items` atomics map.

Comment (wibjorn) [reply site/src/patterns/card.md:67]: Whoops, removed this line. space-between is not a valid value for align-items

File: site/src/patterns/card.md [hunk 12]
```diff
@@ -48,28 +48,86 @@ This card type contains a super title labelled by the card's type, a title, an i
...
-				<p class="color-text-subtle margin-bottom-xs">Videos to help you get started.</p>
-				<a class="text-decoration-none stretched-link" href="#">Watch videos</a>
+				<p class="font-size-xl font-weight-semibold margin-bottom-xxs">Full Name</p>
+				<p href="#">Role</p>
 			</div>
 		</div>
+		<p class="margin-block-xs color-text-subtle">
+			"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor."
+		</p>
+		<a href="#" class="display-block text-decoration-none margin-top-auto font-size-md"
```
Comment (ollips) [inline site/src/patterns/card.md:128]: It keeps bothering me to not see the underline when I'm hovering over the link. I also see that in docs all anchor tags don't have the underline on the default case, which makes `text-decoration-none` unnecessary here, but it is in the atlas. Should it be synced?

**Comment by wibjorn** [comment]
Change set for margin change

**Review by homing1 — APPROVED** [review]

**Review by BenDMyers — APPROVED** [review]

**Review by ollips — APPROVED** [review]

**Pull request merged**